doc/sphinx/proofs/automatic-tactics/logic.rst

```diff
@@ -72,105 +76,88 @@ Therefore, the use of :tacn:`intros` in the previous proof is unnecessary.
+      types with one constructor and no indices, i.e. record-style connectives.
 
 .. flag:: Intuition Negation Unfolding
 
    Controls whether :tacn:`intuition` unfolds inner negations which do not need
-   to be unfolded. This flag is on by default.
+   to be unfolded. The flag is on by default.
 
 .. tacn:: rtauto
-   :name: rtauto
 
-   The :tacn:`rtauto` tactic solves propositional tautologies similarly to what
-   :tacn:`tauto` does. The main difference is that the proof term is built using a
+   Solves propositional tautologies similarly to
+   :tacn:`tauto`, but the proof term is built using a
    reflection scheme applied to a sequent calculus proof of the goal.  The search
    procedure is also implemented using a different technique.
 
-   Users should be aware that this difference may result in faster proof-search
-   but slower proof-checking, and :tacn:`rtauto` might not solve goals that
+   Users should be aware that this difference may result in faster proof search
+   but slower proof checking, and :tacn:`rtauto` might not solve goals that
    :tacn:`tauto` would be able to solve (e.g. goals involving universal
    quantifiers).
 
    Note that this tactic is only available after a ``Require Import Rtauto``.
 
-.. tacn:: firstorder
-   :name: firstorder
+.. tacn:: firstorder {? @ltac_expr } {? using {+, @qualid } } {? with {+ @ident } }
 
-   The tactic :tacn:`firstorder` is an experimental extension of :tacn:`tauto` to
-   first- order reasoning, written by Pierre Corbineau. It is not restricted to
-   usual logical connectives but instead may reason about any first-order class
-   inductive definition.
+   :token:`ltac_expr`
+     Tries to solve the goal with :token:`ltac_expr` when no logical rule applies.
+     If unspecified, the tactic uses the default from the :opt:`Firstorder Solver`
+     option.
 
-.. opt:: Firstorder Solver @tactic
-   :name: Firstorder Solver
+   :n:`using {+, @qualid }`
+     Adds the lemmas :n:`{+, @qualid }` to the proof search environment. If :n:`@qualid`
+     refers to an inductive type, its constructors are
+     added to the proof search environment.
 
-   The default tactic used by :tacn:`firstorder` when no rule applies is
-   :g:`auto with core`, it can be reset locally or globally using this option.
+   :n:`with {+ @ident }`
+     Adds lemmas from :tacn:`auto` hint bases :n:`{+ @ident }` to the proof search
+     environment.
 
-   .. cmd:: Print Firstorder Solver
-
-      Prints the default tactic used by :tacn:`firstorder` when no rule applies.
-
-.. tacv:: firstorder @tactic
-
-   Tries to solve the goal with :n:`@tactic` when no logical rule may apply.
-
-.. tacv:: firstorder using {+ @qualid}
-
-   .. deprecated:: 8.3
-
-      Use the syntax below instead (with commas).
+   An experimental extension of :tacn:`tauto` to
+   first-order reasoning. It is not restricted to
+   usual logical connectives but instead can reason about any first-order class
+   inductive definition.
```

Comment:
Sorry, I may have failed to say this in my previous review, but this short description should appear above the description of the various optional clauses.